### ISSUE TITLE
Merging type_q into one Q results in "TypeError: can't pickle dict_values objects"

### ISSUE THREAD

**Issue by tm-kn** [wagtail/wagtail#4489]
### Issue Summary

When trying to merge `type_q` queries into one Q I get an error saying.
```
TypeError: can't pickle dict_values objects
```

### Steps to Reproduce

```python
from django.db.models import Q

from wagtail.core.models import Page

from home.models import HomePage


type_q =Page.objects.type_q(HomePage)
query = Q()
query |= type_q
```

Probably can be solved converting `dict_values` to a `list`.
https://github.com/wagtail/wagtail/blob/1849f0d54a16432eda2471fda5a602a26d5a5740/wagtail/core/query.py#L175-L179

### Technical details

* Python version: 3.6.4
* Django version: 2.0.3
* Wagtail version: 2.0
* Browser version: n/a

**Comment by nott** [comment]
I'm going to work on this during the sprint in Minsk

**Comment by gasman** [comment]
@nott This has already been addressed by #4814 - it's just waiting for one of us to review it (and if you want to sanity-check that PR, that would be good too :-) )

**Comment by nott** [comment]
@gasman great
i'll take a look

**Comment by gasman** [comment]
Fixed in #4814
